### PR TITLE
plot_spectra heatmap should take kwarg as well

### PR DESCRIPTION
**Pull request by HanHsuanWu**
As title, currently plot_spectra heatmap does not take kwarg for .plot().

### CODE REVIEW

**Review by ericpre — COMMENTED** [review]
The docstrings on `**kwargs` needs to be updated, for now, it says "Has no effect on 'heatmap' style."
It also needs a changelog entry - this is useful to inform user of change of behaviour but also for developer to keep track when changes have implemented.

Can you rebase the changes on the `RELEASE_next_major` branch, because we are aiming at releasing hyperspy 2.0 mid-Sept. and working on that branch will avoid us to merge the `RELEASE_next_minor` into the `RELEASE_next_major`.

**Comment by HanHsuanWu** [comment]
Hi Eric,

I change the doc for plot_spectra and added a changelog entry.
I have rebased to next_major and changed the commit to next_major.

File: hyperspy/drawing/utils.py [hunk 1]
```diff
@@ -1392,7 +1392,6 @@ def plot_spectra(
     **kwargs, optional
         Keywords arguments passed to :py:func:`matplotlib.pyplot.figure` or
         :py:func:`matplotlib.pyplot.subplots` if style='mosaic'.
```
Comment (ericpre) [inline hyperspy/drawing/utils.py]:
```suggestion
    **kwargs : dict, optional
        Depending on the style used, the keyword arguments are passed to different functions

        - ``"overlap"`` or ``"cascade"``: arguments passed to :py:func:`matplotlib.pyplot.figure`
        - ``"mosiac"``: arguments passed to :py:func:`matplotlib.pyplot.subplots`
        - ``"heatmap"``: arguments  passed to :py:meth:`~.api.signals.Signal2D.plot`.
```

**Review by ericpre — COMMENTED** [review]
Suggested clarifications/more details for the docstring.

**Review by ericpre — APPROVED** [review]

**Pull request merged**